### PR TITLE
[luci] Revise shape inference with shape_status

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise shape inference with shape_status
- shape inference will fail for some cases when shape is none
- for this just use shape given from the file

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #1610, from draft #1363

**Comment by seanshpark** [comment]
@mhs4670go @llFreetimell @jinevening PTAL

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 1]
```diff
@@ -2041,11 +2041,30 @@ bool CircleShapeInferenceRule::recognize(const loco::Dialect *d) const
...
+
+  if (is_shape_undefined)
+    shape = circle_node->accept(&alg);
+  else
+  {
+    if (is_shape_none || is_scalar)
+      shape = own_shape(circle_node);
+    else
+      shape = circle_node->accept(&alg);
+  }
```
Comment (mhs4670go) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:2063]:
```suggestion
  if (is_shape_none || is_scalar)
    shape = own_shape(circle_node);
  else
    shape = circle_node->accept(&alg);
```
(Optional )How about this change?

Comment (seanshpark) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:2063]: `is_scalar` can be true when `is_shape_undefined` is true

**Pull request merged**